pkg/logger/logger.go

```diff
@@ -0,0 +1,99 @@
+package logger
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"strings"
+	"sync"
+)
+
+const (
+	ConsoleEncode        = "console"
+	initializingLogError = "initializing logger error: "
+)
+
+var (
+	baseLogger *zap.Logger
+	lock       sync.RWMutex
+)
+
+type Config struct {
+	Level    string `json:"level"`
+	Encoding string `json:"encoding"`
+}
+
+type Field struct {
+	Key   string
+	Value interface{}
+}
+
+type Logger struct {
+	logger *zap.Logger
+}
+
+var stringToLevel = map[string]zapcore.Level{
+	"debug": zapcore.DebugLevel,
+	"info":  zapcore.InfoLevel,
+	"warn":  zapcore.WarnLevel,
+	"error": zapcore.ErrorLevel,
+	"fatal": zapcore.FatalLevel,
+	"panic": zapcore.PanicLevel,
+}
+
+func InitLogger(config Config) {
+	lock.Lock()
+	defer lock.Unlock()
+
+	baseLogger = configureAndBuildLogger(config)
+
+	zap.ReplaceGlobals(baseLogger)
+}
+
+func NewLogger(config ...Config) *Logger {
+	lock.Lock()
+	defer lock.Unlock()
+
+	var cfg Config
+	if len(config) > 0 {
+		cfg = config[0]
```

Comment:
Why do we allow more than 1 config argument, if we are not taking them into account later?

Reply:
This is how Go handles optional parameters. In this way, we can call the method logger.NewLogger() or logger.NewLogger(config). The constructor was created to have it in case we need it, but in reality, we use the init function